### ISSUE TITLE
sudo causing problems

### ISSUE THREAD

**Issue by nsturnitin** [maruf-freelancer/capistrano-deploy-lock#1]
line 187 and 192 contain elevation which is throwing an exception in our secured environments.  The app user we use for deployments does not have sudoerers priv so we get error:

sudo %{touch #{fetch(:deploy_lock_file)}}

We even put a workaround in place so the task is fired from a different user:

utils.rb:
def UD.custom_ssh_options_around task
    ssh_options = {}
    yield ssh_options

```
before task, "#{task}_custom_ssh_options_before" do
  @orig_ssh_options = fetch :ssh_options
  SSHKit.config.backend.configure do |config|
    config.ssh_options.merge!(ssh_options)
  end
end

after task, "#{task}_custom_ssh_options_after" do
  SSHKit.config.backend.configure do |config|
    config.ssh_options = @orig_ssh_options
  end
end
```

  end

deploy.rake:
UD.custom_ssh_options_around 'deploy:create_lock' do |options|
    options[:user] = ENV['USER']
  end

  UD.custom_ssh_options_around 'deploy:unlock:default' do |options|
    options[:user] = ENV['USER']
  end

  UD.custom_ssh_options_around 'deploy:unlock:force' do |options|
    options[:user] = ENV['USER']
  end

The sudo command worked, but then it fails on the following line when it uploads the .yaml file because it is owned by root.

Suggested workaround:
Remove line 187, or just remove the sudo on that line.

Will you be able to do this quickly?

**Comment by nsturnitin** [comment]
guess not

**Comment by mhbcse** [comment]
Sorry I have not seen your comment. I will solve this today.

**Comment by nsturnitin** [comment]
I ended up just removing the entire line since it's redundant:

def write_deploy_lock(deploy_lock)
  upload! StringIO.new(deploy_lock.to_yaml), fetch(:deploy_lock_file)
end

**Comment by mhbcse** [comment]
Thanks for your valuable input. Solved the problem in version 1.0.3
